dali/test/python/test_dali_variable_batch_size.py

```diff
@@ -372,18 +445,294 @@ def pipe_no_input(max_batch_size, input_data, device):
     run_pipeline(generate_data(31, 13, image_like_shape_generator), pipeline_fn=pipe_no_input)
 
-def test_constant():
+
+def no_input_op_helper(operator_fn, opfn_args={}):
+    check_pipeline(
+        generate_data(31, 13, image_like_shape_generator, lo=0, hi=255, dtype=np.uint8),
+        pipeline_fn=single_op_pipeline, input_layout="HWC", operator_fn=operator_fn, needs_input=False, **opfn_args)
+
+
+no_input_ops = [
+    (fn.constant, {'fdata': 3.1415, 'shape': (10, 10)}),
+    (fn.transforms.translation, {'offset': (2, 3), 'devices': ['cpu']}),
+    (fn.transforms.scale, {'scale': (2, 3), 'devices': ['cpu']}),
+    (fn.transforms.rotation, {'angle': 30.0, 'devices': ['cpu']}),
+    (fn.transforms.shear, {'shear': (2., 1.), 'devices': ['cpu']}),
+    (fn.transforms.crop, {'from_start': (0., 1.), 'from_end': (1., 1.), 'to_start': (0.2, 0.3), 'to_end': (0.8, 0.5), 'devices': ['cpu']}),
+]
+
+
+def test_no_input_ops():
+    for op, args in no_input_ops:
+        yield no_input_op_helper, op, args
+
+
+def test_combine_transforms():
     def pipe(max_batch_size, input_data, device):
         pipe = Pipeline(batch_size=max_batch_size, num_threads=4, device_id=0)
-        # just to drive the variable batch size.
-        batch_size_setter = fn.external_source(source=input_data, cycle=False, device=device)
-        data = fn.constant(fdata=3.1415, shape=(10, 10), device=device)
-        pipe.set_outputs(data, batch_size_setter)
+        with pipe:
+            # just to drive the variable batch size.
+            batch_size_setter = fn.external_source(source=input_data, cycle=False, device=device)
+            t = fn.transforms.translation(offset=(1, 2))
+            r = fn.transforms.rotation(angle=30.0)
+            s = fn.transforms.scale(scale=(2, 3))
+            out = fn.transforms.combine(t, r, s)
+        pipe.set_outputs(out, batch_size_setter)
         return pipe
 
     check_pipeline(
         generate_data(31, 13, custom_shape_generator(2, 4), lo=1, hi=255, dtype=np.uint8),
-        pipeline_fn=pipe)
+        pipeline_fn=pipe, devices=['cpu'])
+
+
+def test_dl_tensor_python_function():
+    def dl_tensor_operation(tensor):
+        tensor = torch_dlpack.from_dlpack(tensor)
+        tensor_n = tensor.double() / 255
+        ret = tensor_n.sin()
+        ret = torch_dlpack.to_dlpack(ret)
+        return ret
+
+    def batch_dl_tensor_operation(tensors):
+        out = [dl_tensor_operation(t) for t in tensors]
+        return out
+
+    def pipe(max_batch_size, input_data, device, input_layout=None):
+        pipe = Pipeline(batch_size=max_batch_size, num_threads=4, device_id=0,
+                        exec_async=False, exec_pipelined=False)
+        with pipe:
+            input = fn.external_source(source=input_data, cycle=False, device=device,
+                                       layout=input_layout)
+            output = fn.dl_tensor_python_function(input, function=batch_dl_tensor_operation, batch_processing=True)
```

Comment:
Would it make sense to test non-batch variant?

Reply:
Done